src/UbiHttp.cpp

```diff
@@ -0,0 +1,405 @@
+    return ERROR_VALUE;
+  }
+
+  uint16_t pathLength = _pathLength(device_label, variable_label);
+  char *path = (char *)malloc(sizeof(char) * pathLength + 1);
+  sprintf(path, "/api/v1.6/devices/%s/%s/lv", device_label, variable_label);
+
+  if (_debug) {
+    Serial.print(F("Requesting to URL: "));
+    Serial.println(path);
+  }
+
+  uint16_t requestLineLength = _requestLineLength(path);
+  char *message = (char *)malloc(sizeof(char) * requestLineLength + 1);
+  sprintf(message,
+          "GET %s HTTP/1.1\r\nHost: %s\r\nX-Auth-Token: "
+          "%s\r\nUser-Agent: %s\r\nContent-Type: "
+          "application/json\r\nConnection: close\r\n\r\n",
+          path, _host, _token, _user_agent);
+
+  if (_debug) {
+    Serial.println(F("Request sent"));
+    Serial.println(message);
+  }
+
+  _client_https_ubi.print(message);
+
+  while (_client_https_ubi.connected()) {
+    if (_client_https_ubi.available()) {
+      String line = _client_https_ubi.readStringUntil('\n');
+      if (line == "\r") {
+        break;
+      }
+    }
+  }
+
+  free(message);
+  free(path);
+
+  double value = _parseServerAnswer();
+
+  _client_https_ubi.flush();
+  _client_https_ubi.stop();
+  return value;
+}
+
+double UbiHTTP::_parseServerAnswer() {
+  /**
+   * @param _charResponseLength 3 is the maximun amount of digits for the length
+   * to have
+   */
+  char *_charLength = (char *)malloc(sizeof(char) * 3);
+
+  _parsePartialServerAnswer(_charLength);
+  /**
+   * The server respond the length of the value in HEX so it has to be converted
+   * to DEC
+   * */
+  uint8_t length = UbiUtils::hexadecimalToDecimal(_charLength);
+  char *_charValue = (char *)malloc(sizeof(char) * length + 1);
+
+  _parsePartialServerAnswer(_charValue);
+
+  double value = strtof(_charValue, NULL);
+
+  free(_charLength);
+  free(_charValue);
+
+  return value;
+}
+
+/**
+ * @brief Calculate the lenght of the request line to be send over HTTP to the
+ * server
+ *
+ * @param path address of the endpoint to gather the data
+ * @return uint16_t  Lenght of the request line
+ */
+uint16_t UbiHTTP::_requestLineLength(char *path) {
+  uint16_t endpointLength = strlen(
+                                "GET  HTTP/1.1\r\nHost: \r\nX-Auth-Token: "
+                                "\r\nUser-Agent: \r\nContent-Type: "
+                                "application/json\r\nConnection: close\r\n\r\n") +
+                            strlen(path) + strlen(_host) + strlen(_token) + strlen(_user_agent);
+  return endpointLength;
+}
+
+/**
+ * @brief Calculate the lenght of the path to be send over HTTP to the server
+ *
+ * @param device_label device label of the device
+ * @param variable_label variable label to be updated or fetched
+ * @return uint16_t  Lenght of the endpoint path
+ */
+uint16_t UbiHTTP::_pathLength(const char *device_label, const char *variable_label) {
+  uint16_t endpointLength = strlen("/api/v1.6/devices///lv") + strlen(device_label) + strlen(variable_label);
```

Comment:
@jotathebest the URL has three `/`, is this correct?

Reply:
> @jotathebest the URL has three `/`, is this correct?

@woakas It's correct, it shoud be like this `/api/v1.6/devices/{device_label}/{variable_labe}/lv`

Reply:
Yes, it is wrong. I did not notice it because we have the same error at the library for ESP8266, I am not sure why it works ...

Reply:
Now I understand this line's purpose, @crisap94 at the ESP8266 made in this way the script because you just need to measure the number of bytes of the entire path, so it does not matter the order of the chars as always as you keep all of them. If you replace the line by this:

`uint16_t endpointLength = strlen("/api/v1.6/devices///lv") + strlen(device_label) + strlen(variable_label);`

you will get a runtime error at the uc.